### PR TITLE
Precompile callback before performing a timed test.

### PR DESCRIPTION
**Pull request by maleadt**
Ref #12344

### CODE REVIEW

File: test/parallel_exec.jl [hunk 1]
```diff
@@ -397,14 +397,17 @@ workloads = hist(@parallel((a,b)->[a;b], for i=1:7; myid(); end), nprocs())[2]
     rr2 = Channel()
     rr3 = Channel()
 
+    callback() = all(map(isready, [rr1, rr2, rr3]))
+    # precompile functions which will be tested for execution time
+    @test !callback()
+    @test timedwait(callback, 0.0) == :timed_out
```
Comment (yuyichao) [inline test/parallel_exec.jl]: Maybe `===` instead

**Comment by yuyichao** [comment]
LGTM

File: test/parallel_exec.jl (hunk 1 above)
Comment (maleadt) [inline test/parallel_exec.jl]: Okay, changed.

**Comment by yuyichao** [comment]
I'm wondering if you've managed to make the win64 freeze more reproducible (which would be nice).

**Comment by tkelman** [comment]
We should probably do the same on this `@threadcall` test: https://github.com/JuliaLang/julia/blob/231978501df8848ad4d6e3833955caf5e49c1485/test/ccall.jl#L229

**Pull request merged**